### ISSUE TITLE
[Feature-Request] Support for `type`

### ISSUE THREAD

**Issue by bennet-esyoil** [kimjbstar/prisma-class-generator#21]
In MongoDB we use the Composite-Types `type`-Declaration (https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#defining-composite-types), which enables us to use nested-documents.

These are currently not being transformed into Classes, which means that classes which use them break. 

## Expected Behavior
Also generate a class for the types - and import them in the entity.

## Actual Behavior
Does not generate classes, which leads to missing imports and breaks building.

## Example Schema-File:
```prisma
model Dealer {
  id         String    @id @default(auto()) @map("_id") @db.ObjectId
  createdAt  DateTime? @default(now())
  modifiedAt DateTime  @updatedAt
  name       String
  slug       String    @unique
  address    Address
  Offer      Offer[]

  @@map("dealers")
}

type Address {
  city         String
  postalCode   String
  street       String
  streetNumber String
}
```

Build result would be:
```ts
import { Offer } from './offer';
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';

export class Dealer {
  @ApiProperty({ type: String })
  id: string = undefined;

  @ApiPropertyOptional({ type: Date })
  createdAt?: Date = undefined;

  @ApiProperty({ type: Date })
  modifiedAt: Date = undefined;

  @ApiProperty({ type: String })
  name: string = undefined;

  @ApiProperty({ type: String })
  slug: string = undefined;

  @ApiProperty()
  address: Address = undefined;

  @ApiProperty({ isArray: true, type: () => Offer })
  Offer: Offer[] = undefined;
}
```

## Specifications

-   Version: `"prisma-class-generator": "^0.2.4"`
-   Prisma Version: `"@prisma/client": "^4.7.1"`
-   Platform: linux

**Comment by ArkasDev** [comment]
Unfortunately, the generator is completely unusable for me without supporting type.

**Comment by matmar10** [comment]
Our team would also benefit from this support. Right now we're using the generator one time, and then manually editing the code to change the annotations.

Adding type support would be a huge win.

**Comment by kimjbstar** [comment]
fixed by #49
